task/build-maven-zip/0.1/build-maven-zip.yaml

```diff
@@ -0,0 +1,210 @@
+  - default: "generic"
+    description: The prefetched content which is used in the build.
+    name: PREFETCH_INPUT
+    type: string
+  - default: "maven-repository"
+    description: The root of the prefetched artifacts
+    name: PREFETCH_ROOT
+    type: string
+  - default: "maven-repository"
+    description: The zip bundle name of archived artifacts
+    name: BUNDLE_NAME
+    type: string
+  - default: ""
+    description: Delete image tag after specified time. Empty means to keep the image tag. Time values could be something like 1h, 2d, 3w for hours, days, and weeks, respectively.
+    name: IMAGE_EXPIRES_AFTER
+    type: string
+  - name: caTrustConfigMapName
+    type: string
+    description: The name of the ConfigMap to read CA bundle data from.
+    default: trusted-ca
+  - name: caTrustConfigMapKey
+    type: string
+    description: The name of the key in the ConfigMap that contains the CA bundle data.
+    default: ca-bundle.crt
+
+  results:
+  - description: Digest of the image just built
+    name: IMAGE_DIGEST
+  - description: Image repository and tag where the built image was pushed
+    name: IMAGE_URL
+  - name: SBOM_BLOB_URL
+    description: Reference of SBOM blob digest to enable digest-based verification from provenance
+    type: string
+  stepTemplate:
+    volumeMounts:
+      - mountPath: /shared
+        name: shared
+    env:
+    - name: IMAGE
+      value: $(params.IMAGE)
+    - name: INPUT
+      value: $(params.PREFETCH_INPUT)
+    - name: PKG_ROOT
+      value: $(params.PREFETCH_ROOT)
+    - name: BUNDLE_NAME
+      value: $(params.BUNDLE_NAME)
+    - name: IMAGE_EXPIRES_AFTER
+      value: $(params.IMAGE_EXPIRES_AFTER)
+
+  steps:
+  - image: quay.io/konflux-ci/appstudio-utils@sha256:d0fecb67dd98d874704f2e3d20686363de7cbb42c15ad2b1d8e8c38118c50078
+    name: prepare
+    computeResources:
+      limits:
+        memory: 8Gi
+        cpu: '4'
+      requests:
+        memory: 2Gi
+        cpu: '1'
+    script: |
+      #!/bin/bash
+      set -o pipefail
+      # Generate checksums for all maven artifact files. It will ignore the checksum files
+      # and signature files if they existed there
+      pkgpath="$(workspaces.source.path)/cachi2/output/deps/$INPUT"
+      if [ -d "$pkgpath/${PKG_ROOT}" ]; then
+        echo "Generating checksums for artifacts"
+        while IFS= read -r -d '' f
+        do
+          md5sum "$f" | awk '{print $1}'> "$f.md5"
+          sha1sum "$f" | awk '{print $1}'> "$f.sha1"
+          sha256sum "$f" | awk '{print $1}'> "$f.sha256"
+        done <   <(find "$pkgpath/${PKG_ROOT}" -type f ! -name "*.md5" \
+        ! -name "*.sha1" ! -name "*.sha128" ! -name "*.sha256" \
+        ! -name "*.sha512" ! -name "*.asc" -print0)
+        # Bundle the artifacts and checksums together into a zip file
+        cd "$pkgpath" || exit
+        echo "create maven zip to $(workspaces.source.path)/cachi2/output/${BUNDLE_NAME}.zip"
+        zip -rq "${BUNDLE_NAME}.zip" "${PKG_ROOT}"
```

Comment:
The `generic` fetcher isn't just for maven artifacts. Is it ok to bundle all kinds of arbitrary artifacts into the "Maven zip"?

Reply:
The design for this generic-fetcher can support all kinds of "artifacts" here, not maven. But actually maven does not have a strict rule to limit what's maven artifacts, it's also support all kinds of files as "artifacts", so we don't have a rule to distinguish the non-maven artifacts in these arbitrary ones. So this should be only controlled by the user's lockfiles.